SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -70,35 +107,51 @@ function Invoke-lpass {
-
-    $IgnoreErrors = $ErrorActionPreference -eq 'SilentlyContinue'
     $UseNative = -not $UseWSL
 
     if (($UseNative -and -not (Get-Command $lpassPath -EA 0)) -or 
         ($UseWSL -and (& wsl $lpassPath --version ) -notlike 'LastPass CLI*')) {
         throw "lpass executable not found or installed."
     }
 
-    # All other implementations seemed to succeed on 1 or more platform but failed when considering Windows (Wsl) / Linux / Mac together
-    if ($InputObject) {
-        switch ($true) {
-            {$UseNative -and $IgnoreErrors}  {$result = $InputObject | & $lpassPath @Arguments 2>&1; break}
-            {$UseNative}                     {$result = $InputObject | & $lpassPath @Arguments; break}
-            {$UseWSL -and $IgnoreErrors}     {$result = $InputObject | & wsl $lpassPath @Arguments 2>&1; break}
-            {$UseWSL}                        {$result = $InputObject | & wsl $lpassPath @Arguments; break}
-        }
-    } else {
-        switch ($true) {
-            { $UseNative -and $IgnoreErrors }   {$result = & $lpassPath @Arguments 2>&1; break}
-            { $UseNative }                      {$result = & $lpassPath @Arguments; break}
-            { $UseWSL -and $IgnoreErrors }      {$result = & wsl $lpassPath @Arguments 2>&1; break}
-            { $UseWSL }                         {$result = & wsl $lpassPath @Arguments; break}
+    $Params = @{
+        InputObject = $InputObject
+        UseNative   = $UseNative
+        lpassPath   = $lpassPath
+        Arguments   = $Arguments
+    }
+
+    # If we do redirect on the command themselves, it doesn't work.
+    # Doing redirect on the commands wrapped in a function work.
+    
+    if ($Arguments.Count -gt 0) {
+        switch ($Arguments[0]) {
+            'login' {  
+                # We want the prompt always, so no redirect
+                $result = Invoke-lpassInternal @Params
+            }
+            'show' {
+                # We might want the prompt, but are not sure yet.
+                $result = Invoke-lpassInternal @Params 2>&1
+                # If we get the message stating we might be logged out, we reissue the command without redirect (for Prompt)
+                if ($result -is [System.Management.Automation.ErrorRecord] -and [String]$result -like $lpassMessage.LoggedOut) {
+                    $result2 = Invoke-lpassInternal @Params
+                    # If $result2 -eq $null, something will have been printed in the console (because we disabled the redirect)
+                    # We therefore want to keep the original $result "Logged out" so it is thrown later on
+                    # If not $null, we want to evaluate $result2 instead and discard $result
+                    if ($null -ne $result2) {$result = $result2}
+                }
+            }
+            Default {
+                # By default, we always redirect streams
+                $result = Invoke-lpassInternal @Params 2>&1
+            }
```

Comment:
That's the bit  I was talking about.
By default, redirect all.
Login = No redirect ever (used by Connect-LastPass)
Show = Redirect first, check for logged out message and only if that occurs, redo the command without redirect so prompt is allowed.

Reply:
We could even have gone further and do a login if we got $null, meaning the user was likely logged out and it would have mostly worked perfectly except in the scenario where a disconnected user cancelled the password prompt on purpose. In that case, that would do a password prompt then ask you for your user and a second password prompt.